### PR TITLE
Avoid excessive execution of cloud-config

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Update the checksum of the `download-cloud-config.sh` right after it is applied. Otherwise the update will be detected twice and this will restart the systemd services twice. 

**Which issue(s) this PR fixes**:
Follow-up on #3583 

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/operation/botanist/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh [hunk 1]
```diff
@@ -104,7 +104,7 @@ if ! diff "$PATH_CLOUDCONFIG" "$PATH_CLOUDCONFIG_OLD" >/dev/null || ! diff "$PAT
 {{- end }}
     echo "Successfully restarted all units referenced in the cloud config."
     cp "$PATH_CLOUDCONFIG" "$PATH_CLOUDCONFIG_OLD"
-    cp "$PATH_CCD_SCRIPT_CHECKSUM" "$PATH_CCD_SCRIPT_CHECKSUM_OLD"
+    md5sum ${PATH_CCD_SCRIPT} > "$PATH_CCD_SCRIPT_CHECKSUM_OLD"
```
Comment (timuthy) [inline pkg/operation/botanist/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh]: Shouldn't it already be enough to move the line up before the daemons are restarted? This change is somehow hard to understand and brittle for maintenance. WDYT?

Comment (vpnachev) [reply pkg/operation/botanist/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh]: I prefer to keep the update of the `*_OLD` files together.
Also, after the restarts it is guaranteed that all required changes are applied without failures.

Comment (timuthy) [reply pkg/operation/botanist/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh]: Okay, can you then put a small comment inline why copying is not enough like done a line above?

Comment (vpnachev) [reply pkg/operation/botanist/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh]: I was thinking the same, but also considered that this file is downloaded every 30 seconds, thus I decided to not add comment, but seems it is required.

**Review by timuthy — APPROVED** [review]
Thanks @vpnachev 
/lgtm

**Comment by rfranzke** [comment]
ℹ️ I will open a cherry-pick PR for `release-v1.17` once #3603 is merged.

**Pull request merged**